sds_data_manager/lambda_code/SDSCode/upload_api.py

```diff
@@ -73,6 +78,25 @@ def lambda_handler(event, context):
 
     s3_key_path = science_file.construct_upload_path()
 
+    # Check for already existing file in the database
+    with Session(db.get_engine()) as session:
+        # query and check for a matching file path
+        query = select(models.FileCatalog.__table__).where(
+            models.FileCatalog.file_path == s3_key_path
+        )
```

Comment:
Should this be using an exists query?
https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.exists

I'm out of my element on this, so since this works for now we can try and optimize the queries later.

I also thought that the unique constraint might automatically handle these for us too :-/ Maybe @bourque would know more.

Reply:
It seemed like in version 2 of sqlalchemy, they are moving more toward the select().where() syntax and away from the .query() syntax in version 1 (the documentation linked in version 1). It looks like you can append an exists() to where() (`select().where().exists()`) in version 2 though, so I'll give that a shot

Reply:
@bourque Have you used the exists() construct before? It's not working how I expected it to. I thought from the documentation I could do:

```python
query = exists().where(models.FileCatalog.file_path == s3_key_path)
session.execute(query)
```

but it's creating an exists construct instead of sql that I can execute

https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.exists

Reply:
@sdhoyt Unfortunately I haven't used `exists()` before, I think your guess is as good as mine.

RE: unique constraints -- Correct, I think they are only applied when the insertion of data is executed.

Reply:
Ok sounds good. I'll merge this in for now and look into how that works for later. Thanks for the reviews!